src/task-definitions.js

```diff
@@ -0,0 +1,347 @@
+import NotebookTask from './notebook-task'
+import { store } from './store'
+import actions from './actions'
+import { isCommandMode,
+  viewModeIsEditor,
+  getCellBelowSelectedId,
+  getCellAboveSelectedId } from './notebook-utils'
+
+const dispatcher = {}
+for (const action in actions) {
+  if (Object.prototype.hasOwnProperty.call(actions, action)) {
+    dispatcher[action] = (...params) => (store.dispatch(actions[action](...params)))
+  }
+}
+
+const oscpu = window.navigator.oscpu || window.navigator.platform
+let OSName = 'Unknown OS'
+if (oscpu.indexOf('Win') !== -1) OSName = 'Windows'
+if (oscpu.indexOf('Mac') !== -1) OSName = 'MacOS'
+if (oscpu.indexOf('X11') !== -1) OSName = 'UNIX'
+if (oscpu.indexOf('Linux') !== -1) OSName = 'Linux'
+
+function commandKey(key) {
+  let ctr = 'Ctrl '
+  if (OSName === 'MacOS') {
+    ctr = '⌘ '
+  }
+  return ctr + key
+}
+
+const TASKS = {}
+
+TASKS.EvaluateCell = new NotebookTask({
+  title: 'Evaluate Cell',
+  keybindings: ['mod+enter'],
+
+  keybindingCallback() {
+    dispatcher.changeMode('command')
+    dispatcher.evaluateCell()
+  },
+})
+
+TASKS.EvaluateAndSelectBelow = new NotebookTask({
+  title: 'Evaluate and Select Below',
+  keybindings: ['shift+enter'],
+
+  keybindingCallback() {
+    if (viewModeIsEditor()) {
+      dispatcher.changeMode('command')
+      dispatcher.evaluateCell()
+      const cellBelowId = getCellBelowSelectedId()
+      if (cellBelowId !== null) {
+        dispatcher.selectCell(cellBelowId, true)
+      } else {
+      // if cellBelowId *is* null, need to add a new cell.
+        dispatcher.addCell('javascript')
+        dispatcher.selectCell(getCellBelowSelectedId(), true)
+      }
+    }
+  },
+})
+
+TASKS.MoveCellUp = new NotebookTask({
+  title: 'Move Cell Up',
+  displayKeybinding: '\u21E7 \u2191',
```

Comment:
Rather than having `displayKeybinding` and `keybindings`, which could accidentally become out of sync, would it make sense to have a function that converts the "ascii" `keybindings` value into Unicode automatically (much as is already done for the Macintosh ⌘ symbol in `commandKey` above?

Also, should we have a preference for actual codepoints rather than `\u` escaped ones here?  I've always felt that `\u` escapes are only desirable if you have to use editors or tools that don't handle unicode well, but this is 2018... :)

Reply:
100%.

Reply:
iirc the material ui components don't play nicely with codepoints, which is why I went with these.